ttnn/cpp/ttnn/tensor/tensor_layout.cpp

```diff
@@ -0,0 +1,208 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "tensor_layout.hpp"
+#include <variant>
+#include <vector>
+#include "ttnn/tensor/enum_types.hpp"
+#include "ttnn/tensor/tensor_impl.hpp"
+#include "ttnn/tensor/tensor_utils.hpp"
+#include "types.hpp"
+
+namespace tt::tt_metal {
+
+namespace {
+size_t round_up(size_t value, size_t multiple) {
```

Comment:
I believe there's another implementation of `round_up` already present in the codebase, it would be nice to reuse it